TodoApi/appsettings.Development.json

```diff
@@ -1,24 +1,52 @@
 {
-  "Serilog": {
-    "MinimumLevel": {
-      "Default": "Information",
-      "Override": {
-        "Microsoft": "Information",
-        "Microsoft.Hosting.Lifetime": "Information"
-      }
+    "Serilog": {
+        "Using": [ "Serilog.Sinks.Seq", "Serilog.Sinks.Elasticsearch" ],
+        "MinimumLevel": {
+            "Override": {
+                "Microsoft": "Information"
+            }
+        },
+        // Available Sinks (aka. Destinations to recieve logging data): https://github.com/serilog/serilog/wiki/Provided-Sinks
+        "WriteTo": [
+            {
+                "Name": "Console",
+                "Args": {
+                    "OutputTemplate": "{Timestamp:yyyy-MM-dd HH:mm:ss.fff} {Level} - {Message:lj}{NewLine}{Exception}"
+                }
+            },
+            // REF: https://github.com/datalust/serilog-sinks-seq
+            {
+                "Name": "Seq",
+                "Args": {
+                    "serverUrl": "http://localhost:8081",
```

Comment:
What happens if seq isn't running? What does this do?

Reply:
"Nothing / No-op".
- No errors are reported. Things just keep ticking over. 🍰 
- `SelfLog.Enable(..)` is "on", then the errors are displayed. 🔥 

Same goes if the settings for the sink are bad. Like for example, the `serverUrl = https://localhost:8888` and that is the wrong port ... the system keeps on working normally.

_This is the same outcome if you hardcode the settings and that data is also bad. (aka. the current `main` branch, as of right now)_

## 🎈 🦄 Experiment

Lets do an experiment to get some data/proof!

First, lets run the three docker instances with their 'default' port/settings, etc.

![image](https://user-images.githubusercontent.com/899878/209760228-7c051a5b-85bd-46c5-a130-83b3373dd6d2.png)

![image](https://user-images.githubusercontent.com/899878/209760520-deead3db-1d7d-4f07-bc09-90aee94ad3d2.png)


but this time, I'll put a bad SEQ port, in the settings:

![image](https://user-images.githubusercontent.com/899878/209760491-77c36680-7e6a-48ee-b518-6cfabcf85029.png)

and i'll make sure SelfLog is not set/commented out/off:

![image](https://user-images.githubusercontent.com/899878/209761147-53af7433-ec24-4645-b52c-bc288ada9c91.png)

and then goto an endpoint which gets auto logged:

![image](https://user-images.githubusercontent.com/899878/209761108-d3b69ce5-a88b-42d3-bd9a-505e3a4ad86c.png)

Kewl!!! _Looks like nothing bad has happened_ .... Now lets turn on SelfLog ...

and 5 seconds into starting up, the background SEQ stuff, throws an exception _which we can now see_

![image](https://user-images.githubusercontent.com/899878/209761228-971ee28a-8d47-4cc2-bc72-c1b3b0bcf096.png)

But! I can still goto the endpoint ok (note the previous seq exception, in the logs still) :

![image](https://user-images.githubusercontent.com/899878/209761307-dc90293f-05c1-4cdd-96d9-ac92b58d2956.png)

Ok kewl.

What about `main` branch? lets see what that does, compared to this ...

So i'll switch back to `main` and then add the `SelfLog` with that turned on:

![image](https://user-images.githubusercontent.com/899878/209761546-ef297944-d88c-4b1b-a97b-dd72b8f227f2.png)

![image](https://user-images.githubusercontent.com/899878/209761804-120f127d-e5ed-468c-91d7-d90be5f913d0.png)

and the results:

![image](https://user-images.githubusercontent.com/899878/209761870-7af563f1-1e6d-4101-80da-8eb8508f5d15.png)


the same as before. So with `SelfLog.Enable` turned "on", we can see the secret mistakes occurring behind the scenes.

Reply:
So if you are not planning on running Seq or Elasticsearch, I would delete those child-section.

I would also have all of these child-sections in the `appsettings.development.json` file, but comment out the `seq` and `elasticsearch` child-sections, just leaving the `console` sink active.

This shows the developers their options but doesn't force them into those AND does mean those sinks are running _by default_ and silently having errors in the background if the developer never wants to run those 'sinks'. That's my opinion on this.

Reply:
Then the section should be deleted in this PR 😄 and it should all be in documentation. Today nothing fails because it isn't configured by default when running locally. When you run with docker compose or you set those config values (for e.g. via env variables) it lights up.

Reply:
Would you consider having it commented out in the file, as the form of documentation ?

Reply:
It should be in the readme. JSON comments are only half supported (look at how poorly they show up on github).

Reply:
🔧